client/src/core/props-behavior.js

```diff
@@ -131,107 +131,148 @@ class PropsSelector {
-            this.engine3d.getSelectableNodes(), true,
+  /**
+  * Select all props in the same chunk within a given radius.
+  *
+  * @param {Object3D} rootProp - Root prop used as center point.
+  * @param {number} radius - Distance (in meters) to include nearby props.
+  */
+  selectNearbyProps(baseProp) {
+    const chunkHandle = baseProp.userData.chunkNodeHandle;
+    if (!chunkHandle) return;
+
+    const chunkId = this.worldManager.chunkKeys.get(chunkHandle);
+    if (!chunkId) return;
+
+    // Find all props in the same chunk
+    // eslint-disable-next-line
+    for (const [propId, obj3d] of this.worldManager.props.entries()) {
+      if (
+        obj3d.userData.chunkNodeHandle === chunkHandle &&
+        obj3d !== baseProp &&
+        obj3d.visible
+      ) {
+        const alreadySelected = this.props.find(
+            ({stagingProp}) =>
+              stagingProp.userData.prop.id === obj3d.userData.prop.id,
         );
-
-    for (const intersect of intersects) {
-      if (intersect.object.parent &&
-          intersect.object.parent.name == pageAssetName) {
-        // Some terrain page has been selected, which is the equivalent
-        // of clicking into the void as far as the props-selector
-        // is concerned
-        if (!add) {
-          // If we're not in multiprop selection mode:
-          // commit the current content to the server
-          this.commitAndClear(false);
+        if (!alreadySelected) {
+          this.selectSingleProp(obj3d);
         }
-
-        done = true;
-        break;
       }
+    }
+  }
 
-      if (validateIntersect(intersect, this.maxCastingDistance)) {
-        // If the object was already selected: nothing to be done...
-        const foundPropId = this.props.findIndex(({stagingProp}) => {
-          return intersect.object.id === stagingProp.id;
-        });
-
-        if (foundPropId >= 0) {
-          if (add) {
-            // ... unless multiprop selection is on, then we deselect this
-            // single object and commit the changes for it.
-            this.commitAndClearSingle(foundPropId, false);
-          }
-          done = true;
-          break;
-        }
-
-        if (!add) {
-          // If we're not in multiprop selection mode:
-          // commit the current content to the server
-          this.commitAndClear(false);
-        }
+  /**
+    * Select a prop for building mode
+    * @param {Vector2} pointer - 2D Pointer for the raycaster.
+    * @param {boolean} add - True to add the newfound prop to the existing
+    *                        selection list, false (default) to clear the
+    *                        list first.
+    * @param {boolean} - True to select nearby objects, false (default) to
+    *                    do a regular object selection
+    */
+  select(pointer, add = false, multi = false) {
+    const selectionRadius = 1.0;
+    let done = false;
 
-        // We expect the object to have pre-computed bounding box geometry
```

Comment:
Same point about the comments here: how and where this logic has been shifted/transformed? There are some `if (add)` and `if (!add)` blocks than no longer have any human-readable explanations, which I think are much needed when dealing with this sort of logic...